assignment/core/src/main/java/com/assignment/core/services/impl/NodeOperationServiceImpl.java

```diff
@@ -0,0 +1,72 @@
+package com.assignment.core.services.impl;
+
+import com.assignment.core.services.NodeOperationService;
+import org.apache.sling.api.resource.PersistenceException;
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.api.resource.ResourceResolver;
+import org.apache.sling.api.resource.ValueMap;
+import org.osgi.service.component.annotations.Component;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.jcr.RepositoryException;
+import java.util.HashMap;
+import java.util.Map;
+
+@Component(service= NodeOperationService.class)
+public class NodeOperationServiceImpl implements NodeOperationService {
+
+    private static final Logger LOG = LoggerFactory.getLogger(NodeOperationServiceImpl.class);
+
+    /**
+     * Selects node operation based on the 'save' parameter.
+     */
+    @Override
+    public void selectNodeOperation(ResourceResolver resolver, String save, long totalMatches) throws PersistenceException, RepositoryException {
+        if ("true".equalsIgnoreCase(save)) {
+            createNode(resolver, totalMatches);
+        }
+        else {
+            deleteNodes(resolver);
+        }
+        resolver.commit();
```

Comment:
commit in save/ delete methods so that when we call createNode() directly from any other code piece the resolver.commit() is there in the operation.